fix: `_comment_action` rule

guilhas07 opened the pull request:
### Problem:
Comments are defined in https://pkg.go.dev/text/template like so: 
```// Comments
{{/* a comment */}}
{{- /* a comment with white space trimmed from preceding and following text */ -}}
	A comment; discarded. May contain newlines.
	Comments do not nest and must start and end at the
	delimiters, as shown here.
```
### Solution

Change the `_comment_action` rule so a comment always has either the delimiter `{{` or `{{- ` (note the space at the end). Also, every element in the rule should be immediate to each other. Meaning that delimiters can't be alone on a line.

qvalentin commented on make_grammar.js:
```diff
@@ -107,7 +107,16 @@ module.exports = function make_grammar(dialect) {
             _comment_action: ($) =>
-                seq($._left_delimiter, $.comment, $._right_delimiter),
+                choice(
+                    seq(token('{{'), $.comment, token.immediate('}}')),
```
The mixed case is missing:

```gotmpl
{{- /* comment */}}
{{/* comment */ -}}
```

guilhas07 commented:
I noticed this repo has a `.prettierrc` file but the `make_grammar.js` isn't formatted with it. Is it okay with you if after everything looks good I push changes that apply prettier formatting?

qvalentin commented:
> I noticed this repo has a `.prettierrc` file but the `make_grammar.js` isn't formatted with it. Is it okay with you if after everything looks good I push changes that apply prettier formatting?

Sure, you will have to rebase the generated files anyway.

guilhas07 commented on make_grammar.js:
```diff
@@ -435,13 +445,8 @@ module.exports = function make_grammar(dialect) {
-            comment: ($) =>
-                token(
-                    choice(
-                        seq('//', /.*/),
```
Yeah, I think this was probably taken from tree-sitter-go https://github.com/tree-sitter/tree-sitter-go/blob/12fe553fdaaa7449f764bc876fd777704d4fb752/grammar.js#L915-L924

guilhas07 commented:
Applied prettier formatting and removed unused functions/variables.
Additionaly, the new tree-sitter version gives a warning to: https://github.com/ngalaiko/tree-sitter-go-template/blob/4fc529dc37efb57796798c1590ef9bc81c20304f/make_grammar.js#L85
due to being a single token in a `choice`. Is this intentional?

guilhas07 commented:
> > Applied prettier formatting and removed unused functions/variables. Additionaly, the new tree-sitter version gives a warning to:
> > https://github.com/ngalaiko/tree-sitter-go-template/blob/4fc529dc37efb57796798c1590ef9bc81c20304f/make_grammar.js#L85
> > 
> > due to being a single token in a `choice`. Is this intentional?
> 
> I think you can just remove the `choice`.

This triggered some changes in the test case, so I updated the tests with `npx tree-sitter test -u`, which changed a lot of the formatting in the `test/corpus/actions.txt`

qvalentin commented:
> > > Applied prettier formatting and removed unused functions/variables. Additionaly, the new tree-sitter version gives a warning to:
> > > https://github.com/ngalaiko/tree-sitter-go-template/blob/4fc529dc37efb57796798c1590ef9bc81c20304f/make_grammar.js#L85
> > > 
> > > due to being a single token in a `choice`. Is this intentional?
> > 
> > 
> > I think you can just remove the `choice`.
> 
> This triggered some changes in the test case, so I updated the tests with `npx tree-sitter test -u`, which raised a lot of cosmetic changes in `test/corpus/actions.txt`.

The diff is getting very difficult to read, could you remove the cosmetic changes (maybe we can later open a dedicated PR for them)?

guilhas07 commented:
> > > > Applied prettier formatting and removed unused functions/variables. Additionaly, the new tree-sitter version gives a warning to:
> > > > https://github.com/ngalaiko/tree-sitter-go-template/blob/4fc529dc37efb57796798c1590ef9bc81c20304f/make_grammar.js#L85
> > > > 
> > > > due to being a single token in a `choice`. Is this intentional?
> > > 
> > > 
> > > I think you can just remove the `choice`.
> > 
> > 
> > This triggered some changes in the test case, so I updated the tests with `npx tree-sitter test -u`, which raised a lot of cosmetic changes in `test/corpus/actions.txt`.
> 
> The diff is getting very difficult to read, could you remove the cosmetic changes 

Sorry about that. It should be better now.

> maybe we can later open a dedicated PR for them

Maybe that could be done on your tree-sitter bump PR? Running `npx tree-sitter test -u` and it should be done.

guilhas07 commented:
> Thanks for the contributions 😄
> 
> Also feel free to review my open PRs.

Thank you. Will do!
